modules/gke-cluster-standard/variables.tf

```diff
@@ -54,6 +54,19 @@ variable "cluster_autoscaling" {
         integrity_monitoring = optional(bool, true)
         secure_boot          = optional(bool, false)
       })
+      upgrade_settings = optional(object({
```

Comment:
We need to redesign this variable.

`strategy` only has two values with `"SURGE"` being the default, and our approach is to always turn those to booleans to clearly formulate the choice to module users, something like

```hcl
upgrade_settings = optional(object({
  blue_green_strategy = optional(bool, false)
```

But then the choice of strategy defines which other fields are possible, so what we typically do in those cases (health checks in load balancer modules for example) is to have multiple optional objects, and validation to ensure only one is selected, which would also redefine `blue_green_settings`. Something like

```hcl
upgrade_settings = optional(object({
  blue_green = optional(object({
    node_pool_soak_duration = optional(string)
    standard_rollout_policy = optional(object({
      batch_percentage    = optional(number)
      batch_node_count    = optional(number)
      batch_soak_duration = optional(string)
    }))
  })
  surge = optional(object({
    max = optional(number)
    unavailable = optional(number)
  })
})
# add validation rule to ensure only one is present if upgrade settings is defined
```

Then if a user set this in the module

```hcl
upgrade_settings = {}
```

you would internally need to assume surge as default.